mesh_map/src/mesh_map.cpp

```diff
@@ -151,50 +147,106 @@ MeshMap::MeshMap(tf2_ros::Buffer& tf, const rclcpp::Node::SharedPtr& node)
   vertex_colors_pub = node->create_publisher<mesh_msgs::msg::MeshVertexColorsStamped>("~/vertex_colors", rclcpp::QoS(1).transient_local());
   vector_field_pub = node->create_publisher<visualization_msgs::msg::Marker>("~/vector_field", rclcpp::QoS(1).transient_local());
   config_callback = node->add_on_set_parameters_callback(std::bind(&MeshMap::reconfigureCallback, this, std::placeholders::_1));
+
+  save_service = node->create_service<std_srvs::srv::Empty>("~/save_map", [this](
+    const std::shared_ptr<std_srvs::srv::Empty::Request> request,
+    std::shared_ptr<std_srvs::srv::Empty::Response>      response)
+  {
+    saveLayers();
+  });
 }
 
 bool MeshMap::readMap()
```

Comment:
Reading the code that loads the mesh map was a bit confusing with all the different cases and different ways the working/non-working variables can be filled (via user or automatically).
Is there a way to simplify the logic?

Reply:
Perhaps there is a way. But I haven't found it yet :D